services/brig/test/integration/API/MLS.hs

```diff
@@ -106,6 +107,54 @@ testKeyPackageClaim brig = do
       ciUser cid @?= qUnqualified u
       ciClient cid @?= kpbeClient e
 
+testKeyPackageSelfClaim :: Brig -> Http ()
+testKeyPackageSelfClaim brig = do
+  -- setup a user u with two clients c1 and c2
+  u <- userQualifiedId <$> randomUser brig
+  [c1, c2] <- for [0, 1] $ \i -> do
+    c <- createClient brig u i
+    -- upload 3 key packages for each client
+    withSystemTempFile "store.db" $ \store _ ->
+      uploadKeyPackages brig store SetKey u c 3
+    pure c
+
+  -- claim own packages but skip the first
+  do
+    bundle <-
+      responseJsonError
+        =<< post
+          ( brig
+              . paths ["mls", "key-packages", "claim", toByteString' (qDomain u), toByteString' (qUnqualified u)]
+              . queryItem "skip_own" (toByteString' c1)
+              . zUser (qUnqualified u)
+          )
+          <!! const 200 === statusCode
+    liftIO $ Set.map (\e -> (kpbeUser e, kpbeClient e)) (kpbEntries bundle) @?= Set.fromList [(u, c2)]
+
+    -- check that we still have all keypackages for client c1
+    count :: KeyPackageCount <-
+      responseJsonError
+        =<< get
+          ( brig . paths ["mls", "key-packages", "self", toByteString' c1, "count"]
+              . zUser (qUnqualified u)
+          )
+        <!! const 200 === statusCode
+    liftIO $ count @?= 3
+
+  -- if another user sets skip_own, nothing is skipped
+  do
+    u' <- userQualifiedId <$> randomUser brig
+    bundle <-
+      responseJsonError
+        =<< post
+          ( brig
+              . paths ["mls", "key-packages", "claim", toByteString' (qDomain u), toByteString' (qUnqualified u)]
+              . queryItem "skip_own" (toByteString' c1)
+              . zUser (qUnqualified u')
+          )
+          <!! const 200 === statusCode
+    liftIO $ Set.map (\e -> (kpbeUser e, kpbeClient e)) (kpbEntries bundle) @?= Set.fromList [(u, c1), (u, c2)]
```

Comment:
Shouldn't entries for `c2` be absent from the response? I would expect so because there were 3 key packages uploaded for `c2` and they were all claimed in a block with line numbers 121-132.

Reply:
Why were they all claimed? Only *one* key package per client is claimed by each endpoint call.

Reply:
You're right! My mistake.

Maybe another useful assertion in the last block would be on the count of available key packages per client after `u'` claimed a key package.

Reply:
That's covered by the test above, though, isn't it?

Reply:
Are you referring to the block in lines 134-142? If so, I thought the outcome is different when `u'` claims key packages compared to `u` claiming it.

Reply:
Alright, I added the assertion.